bottlerocket-settings-models/settings-models/src/kubernetes/mod.rs

```diff
@@ -96,4 +97,36 @@ pub struct KubernetesSettingsV1 {
     hostname_override: ValidLinuxHostname,
     // Generated in `k8s-1.25+` variants only
     seccomp_default: bool,
+    #[cfg(feature = "nvidia-device-plugin")]
+    device_plugins: K8sDevicePluginSettingsV1,
+}
+
+// Define the DevicePlugin struct
+#[model(impl_default = true)]
```

Comment:
Mind adding tests that the appropriate data parses correctly for these? There are a few examples in `modeled-types/kubernetes`.

It may also be wise to move `NvidiaPluginSettings` to `modeled-types/src/kubernetes.rs` so that they can be more-easily re-used by other Kubernetes settings models. (Sorry, this conflicts with advice I gave you verbally in the past).

You may either need to add a dependency from `modeled-types` on the `model-derive` crate, or remove the `model` macro and replace it with the appropriate set of trait derivations, as well as marking fields as optional.

Reply:
changed as suggested.